### PR TITLE
[harvester doc] add doc of alertmanager

### PR DESCRIPTION
**Pull request by w13915984028**
[harvester doc] add doc of alertmanager [CI SKIP]
Signed-off-by: Jian Wang <w13915984028@gmail.com>

Add alertmanager into Harvester document.

View the final doc directly:
https://deploy-preview-209--harvester-preview.netlify.app/next/monitoring/

Related feature:
https://github.com/harvester/harvester/issues/2517

### CODE REVIEW

File: docs/monitoring/monitoring.md [hunk 1]
```diff
@@ -114,6 +114,152 @@ spec.values.prometheus-node-exporter.resources.requests.memory: 30Mi
...
+https://github.com/harvester/harvester/issues/2760
+
+### View and manage alerts
+
+
+#### From Alertmanager dashboard
+
+From below link, user may visit the `Alertmanager` original dashboard.
+
+> https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-alertmanager:9093/proxy/#/alerts
```
Comment (vickyhella) [inline docs/monitoring/monitoring.md:242]:
```suggestion
You can visit the original dashboard of `Alertmanager` from [here](https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-alertmanager:9093/proxy/#/alerts).
```

File: docs/monitoring/monitoring.md [hunk 2]
```diff
@@ -114,6 +114,152 @@ spec.values.prometheus-node-exporter.resources.requests.memory: 30Mi
...
+
+Overall view of `Alertmanager` dashboard.
```
Comment (vickyhella) [inline docs/monitoring/monitoring.md:244]:
```suggestion
The overall view of the `Alertmanager` dashboard is as follows:
```

File: docs/monitoring/monitoring.md [hunk 3]
```diff
@@ -114,6 +114,152 @@ spec.values.prometheus-node-exporter.resources.requests.memory: 30Mi
...
+
+![](/img/v1.1/monitoring/alertmanager-dashboard.png)
+
+View the details of an alert.
```
Comment (vickyhella) [inline docs/monitoring/monitoring.md:248]:
```suggestion
You can view the details of an alert:
```

File: docs/monitoring/monitoring.md [hunk 4]
```diff
@@ -114,6 +114,152 @@ spec.values.prometheus-node-exporter.resources.requests.memory: 30Mi
...
+
+![](/img/v1.1/monitoring/alert-view-detail.png)
+
+#### From Prometheus dashboard
+
+From below link, user may visit the `Prometheus` original dashboard, which has detailed information of alert.
+
+> https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/
+
+The `Alert` top menu, shows all defined `PrometheusRule`s. With the filter `Inactive`, `Pending`, `Firing`, user can get information quickly.
```
Comment (vickyhella) [inline docs/monitoring/monitoring.md:258]:
```suggestion
The `Alerts` menu in the top navigation bar shows all defined `PrometheusRule`s. You can use the filters `Inactive`, `Pending`, and `Firing` to quickly find the information that you need.
```

File: docs/monitoring/monitoring.md [hunk 5]
```diff
@@ -114,6 +114,152 @@ spec.values.prometheus-node-exporter.resources.requests.memory: 30Mi
...
+    alertmanagerConfig: example
+spec:
+  route:
+    continue: true
+    groupBy:
+    - cluster
+    - alertname
+    receiver: "amc-webhook-reciver"
+  receivers:
+  - name: "amc-webhook-reciver"
```
Comment (vickyhella) [inline docs/monitoring/monitoring.md:184]:
```suggestion
    receiver: "amc-webhook-receiver"
  receivers:
  - name: "amc-webhook-receiver"
```

File: docs/monitoring/monitoring.md [hunk 6]
```diff
@@ -114,6 +114,152 @@ spec.values.prometheus-node-exporter.resources.requests.memory: 30Mi
...
+
+To send the alerts to third party servers, users need to config the `AlertmanagerConfig`.
+
+The WebUI config path is `Monitoring & Logging` -> `Monitoring` -> `Alertmanager Configs`.
+
+Create `Alertmanager Config`, click `Namespace` to select the target namespace from drop down menu.
+
+![](/img/v1.1/monitoring/alertmanager-config-create-1.png)
+
+View the created `Alertmanager Configs`, click one of them to continue config.
```
[the remark this answers is not in the document]
Comment (vickyhella) [reply docs/monitoring/monitoring.md:145]: Using the one we just created as an example will be clearer. Otherwise, it may be confusing.

File: docs/monitoring/monitoring.md (hunk 5 above)
Comment (vickyhella) [reply docs/monitoring/monitoring.md:184]: Please double-check if these are typo

File: docs/monitoring/monitoring.md [hunk 7]
```diff
@@ -114,6 +114,152 @@ spec.values.prometheus-node-exporter.resources.requests.memory: 30Mi
...
+
+View the details of an alert.
+
+![](/img/v1.1/monitoring/alert-view-detail.png)
+
+#### From Prometheus dashboard
+
+From below link, user may visit the `Prometheus` original dashboard, which has detailed information of alert.
+
+> https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/
```
Comment (vickyhella) [inline docs/monitoring/monitoring.md:256]:
```suggestion
You can visit the original dashboard of `Prometheus` from [here](https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/), which offers detailed information about the alert.
```

**Review by vickyhella — CHANGES_REQUESTED** [review]
Just some grammatical and spelling nits. The rest LGTM, thanks!

File: docs/monitoring/monitoring.md [hunk 8]
```diff
@@ -114,6 +114,152 @@ spec.values.prometheus-node-exporter.resources.requests.memory: 30Mi
 
 For versions `<= v1.0.0`, the related path and default value are not specified in the `managedchart rancher-monitoring`, you need to add them accordingly.
 
+## Alertmanager
+
+_Available as of v1.1.0_
+
+`Harvester` has an `Alertmanager` to collect and manage all the alerts happened in the cluster.
```
Comment (divya-mohan0209) [inline docs/monitoring/monitoring.md:121]:
```suggestion
`Harvester` has an `Alertmanager` to collect and manage all the alerts happening in the cluster.
```

File: docs/monitoring/monitoring.md [hunk 9]
```diff
@@ -114,6 +114,152 @@ spec.values.prometheus-node-exporter.resources.requests.memory: 30Mi
...
+
+#### Change resource setting
+
+The resource setting of `Alertmanager` can also be changed. It is shown in previous picture.
+
+#### Config AlertmanagerConfig from WebUI
+
+To send the alerts to third party servers, users need to config the `AlertmanagerConfig`.
+
+The WebUI config path is `Monitoring & Logging` -> `Monitoring` -> `Alertmanager Configs`.
```
[the remark this answers is not in the document]
Comment (divya-mohan0209) [reply docs/monitoring/monitoring.md:139]: I would rephrase this as 
```suggestion
On the WebUI, navigate to `Monitoring & Logging` -> `Monitoring` -> `Alertmanager Configs`.
```

File: docs/monitoring/monitoring.md [hunk 10]
```diff
@@ -114,6 +114,152 @@ spec.values.prometheus-node-exporter.resources.requests.memory: 30Mi
...
+
+# kubectl get alertmanagerconfig -A
+NAMESPACE   NAME          AGE
+default     amc-example   27s
+
+```
+
+#### Example of webhook received alert
+
+The alert sent to an webhook server will be in following format:
```
[the remark this answers is not in the document]
Comment (divya-mohan0209) [reply docs/monitoring/monitoring.md:201]: Modifying Vicky's suggestion a bit.
```suggestion
Alerts sent to the webhook server will be in the following format:
```

File: docs/monitoring/monitoring.md [hunk 11]
```diff
@@ -114,6 +114,152 @@ spec.values.prometheus-node-exporter.resources.requests.memory: 30Mi
...
+'externalURL': 'https://192.168.122.200/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-alertmanager:9093/proxy',
+'version': '4',
+'groupKey': '{}/{namespace="longhorn-system"}:{}',
+'truncatedAlerts': 0
+}
+```
+
+:::note
+
+Different receiver may represent the alert in it's format, please refer the related documents.
```
[the remark this answers is not in the document]
Comment (divya-mohan0209) [reply docs/monitoring/monitoring.md:223]: I would link to the related documents if any within our site.

File: docs/monitoring/monitoring.md (hunk 4 above)
Comment (divya-mohan0209) [reply docs/monitoring/monitoring.md:258]: The `` sign needs to either incorporate the plural form OR you need to rephrase it as below for correctness.

```suggestion
The `Alerts` menu in the top navigation bar shows all defined rules in Prometheus. You can use the filters `Inactive`, `Pending`, and `Firing` to quickly find the information that you need.
```

**Review by divya-mohan0209 — CHANGES_REQUESTED** [review]
Few nits.

File: docs/monitoring/monitoring.md (hunk 1 above)
Comment (w13915984028) [reply docs/monitoring/monitoring.md]: Because the link itself is invalid, user need to copy the link and change the `the-cluser-vip` to real vip address (e.g. 192.168.122.100). Here we need to represent the raw link text for user to copy and modify.

File: docs/monitoring/monitoring.md (hunk 7 above)
Comment (w13915984028) [reply docs/monitoring/monitoring.md]: Similar. Because the link itself is invalid, user need to copy the link and change the `the-cluser-vip` to real vip address (e.g. 192.168.122.100). Here we need to represent the raw link text for user to copy and modify.

File: docs/monitoring/monitoring.md (hunk 8 above)
Comment (w13915984028) [reply docs/monitoring/monitoring.md]: change to `happened/happening`.

File: docs/monitoring/monitoring.md [hunk 12]
```diff
@@ -114,6 +114,149 @@ spec.values.prometheus-node-exporter.resources.requests.memory: 30Mi
...
+
+We have already created a [GithHb issue](https://github.com/harvester/harvester/issues/2760) to track upstream changes. Once the feature is available, `Harvester` will adopt it.
+
+### View and Manage Alerts
+
+#### From Alertmanager Dashboard
+
+You can visit the original dashboard of `Alertmanager` from below link.
+
+> https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-alertmanager:9093/proxy/#/alerts
```
Comment (vickyhella) [inline docs/monitoring/monitoring.md:239]:
```suggestion
You can visit the original dashboard of `Alertmanager` from the link below. Note that you need to replace <cluster-vip> with the actual cluster-vip:

> https://<cluster-vip>/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-alertmanager:9093/proxy/#/alerts
```

File: docs/monitoring/monitoring.md [hunk 13]
```diff
@@ -114,6 +114,149 @@ spec.values.prometheus-node-exporter.resources.requests.memory: 30Mi
...
+
+You can view the details of an alert:
+
+![](/img/v1.1/monitoring/alert-view-detail.png)
+
+#### From Prometheus Dashboard
+
+You can visit the original dashboard of `Prometheus` from below link.
+
+> https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/
```
Comment (vickyhella) [inline docs/monitoring/monitoring.md:253]:
```suggestion
You can visit the original dashboard of `Prometheus` from the link below. Note that you need to replace <cluster-vip> with the actual cluster-vip.

> https://<cluster-vip>/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/
```

File: docs/monitoring/monitoring.md (hunk 12 above)
Comment (vickyhella) [reply docs/monitoring/monitoring.md:239]: Thanks for explaining the links! If so, I think it's better to tell users the need to replace the variable.

**Review by vickyhella — APPROVED** [review]
Modified the link-related statements a little bit. The rest LGTM.

File: docs/monitoring/monitoring.md (hunk 5 above)
Comment (w13915984028) [reply docs/monitoring/monitoring.md]: updated, thanks.

File: docs/monitoring/monitoring.md (hunk 12 above)
Comment (w13915984028) [reply docs/monitoring/monitoring.md]: seems `<cluster-vip>` is not fit here.

image

Comment (vickyhella) [reply docs/monitoring/monitoring.md]: Then maybe we should remove the angle brackets ^^

Comment (w13915984028) [reply docs/monitoring/monitoring.md]: keep `the-cluster-vip` and without `<>` makes CI happy.

**Review by guangbochen — APPROVED** [review]
LGTM and please squash the commits, thanks.

**Review by bk201 — APPROVED** [review]
lgtm!

**Comment by w13915984028** [comment]
@guangbochen squashed and ready to merge now.
